lib/util/mp4_generator.js

```diff
@@ -940,3 +1123,79 @@ shaka.util.Mp4Generator.DREF_ = new Uint8Array([
  * @private {!Uint8Array}
  */
 shaka.util.Mp4Generator.DINF_ = new Uint8Array([]);
+
+/**
+ * @typedef {{
+ *    timescale: number,
+ *    duration: number,
+ *    videoNalus: !Array.<string>,
+ *    data: ?shaka.util.Mp4Generator.Data,
+ *    stream: shaka.extern.Stream
+ * }}
+ *
+ * @property {number} timescale
+ *   The Stream's timescale.
+ * @property {number} duration
+ *   The Stream's duration.
+ * @property {!Array.<string>} videoNalus
+ *   The stream's video nalus.
+ * @property {?shaka.util.Mp4Generator.Data} data
+ *   The stream's data.
+ * @property {shaka.extern.Stream} stream
+ *   The Stream.
+ */
+shaka.util.Mp4Generator.StreamInfo;
+
+/**
+ * @typedef {{
+ *    sequenceNumber: number,
+ *    baseMediaDecodeTime: number,
+ *    samples: !Array.<shaka.util.Mp4Generator.Mp4Sample>
+ * }}
+ *
+ * @property {number} sequenceNumber
+ *   The sequence number.
+ * @property {number} baseMediaDecodeTime
+ *   The base media decode time.
+ * @property {Array.<shaka.util.Mp4Generator.Mp4Sample>} samples
```

Comment:
This doesn't match with the definition above, which says this should be `!Array.<shaka.util.Mp4Generator.Mp4Sample>`. That feels more correct, looking at the code.